api/go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/fluxcd/pkg/apis/meta v0.14.0
 	k8s.io/apiextensions-apiserver v0.24.0
 	k8s.io/apimachinery v0.24.0
-	sigs.k8s.io/controller-runtime v0.11.2
+	sigs.k8s.io/controller-runtime v0.12.1
```

Comment:
Please undo this change, controller-runtime v0.12.1 breaks the logging and has serious bugs!

Reply:
Can you elaborate on the bugs? I personally find "breaks logging" not a very strong arguments to stall further updates. As the further set of changes seems to be good: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.12.0

Reply:
Discussed this further. We need to look into our glue around logging before progressing this update.

Reply:
For further reference, we need https://github.com/kubernetes-sigs/controller-runtime/pull/1911 to be included in a `0.12.x` release.